cirq-core/cirq/circuits/circuit.py

```diff
@@ -1868,9 +1871,14 @@ def transform_qubits(
     def _prev_moment_available(self, op: 'cirq.Operation', end_moment_index: int) -> Optional[int]:
         last_available = end_moment_index
         k = end_moment_index
+        op_control_keys = protocols.control_key_names(op)
+        op_qubits = op.qubits
         while k > 0:
             k -= 1
-            if not self._can_commute_past(k, op):
+            moment = self._moments[k]
+            if moment.operates_on(op_qubits) or any(
```

Comment:
`any` is not the correct thing here. Use `len(...)==0`, or `bool(...)`, or just nothing.